syntax = "proto3";

import "authResponse.proto";
import "contact.proto";
import "message.proto";
import "server.proto";

message FetchContacts {
    string user_id = 1;
}

message FetchDmMessages {
    string user_id = 1;
    string fetch_user_name = 2;
}

message FetchServers {
    string user_id = 1;
}

message FetchChannelMessages {
    string server_id = 1;
    string channel_name = 2;
}

message FetchServerMembers {
    string server_id = 1;
}

message SignInRequest {
    string user_name = 1;
    string user_password = 2;
}

message SignUpRequest {
    string user_name = 1;
    string display_name = 2;
    string user_email = 3;
    string user_password = 4;
}

message FetchResponseContacts {
    repeated Contact response = 1;
}

message FetchResponseServers {
    repeated Server response = 1;
}

message FetchResponseMessages {
    repeated Message response = 1;
}

message Body {
    oneof payload {
        FetchContacts f_contacts = 1;
        FetchDmMessages f_dmMessages = 2;
        FetchServers f_servers = 3;
        FetchChannelMessages f_channelMessages = 4;
        FetchServerMembers f_serverMembers = 5;
        SignUpRequest su_request = 6;
        SignInRequest si_request = 7;
        AuthResponse au_response = 8;
        Contact ct_request = 9;
        Contact ct_response = 10;
        FetchResponseContacts f_contacts_response = 11;
        FetchResponseContacts f_servers_response = 12;
        FetchResponseContacts f_messages_response = 13;
    }
}
